### PR TITLE
Add precise ack timeout module

### PR DESCRIPTION
**Pull request by KaSroka**

### CODE REVIEW

File: src/mac_features/nrf_802154_precise_ack_timeout.c [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+#include <assert.h>
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "nrf_802154_notification.h"
+#include "nrf_802154_procedures_duration.h"
+#include "nrf_802154_request.h"
+#include "timer_scheduler/nrf_802154_timer_sched.h"
+
+#define RETRY_DELAY     500      ///< Procedure is delayed by this time if cannot be performed at the moment.
```
Comment (hubertmis) [inline src/mac_features/nrf_802154_precise_ack_timeout.c]: Add a unit [us] to the comment

File: src/mac_features/nrf_802154_precise_ack_timeout.c [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+#include "nrf_802154_request.h"
+#include "timer_scheduler/nrf_802154_timer_sched.h"
+
+#define RETRY_DELAY     500      ///< Procedure is delayed by this time if cannot be performed at the moment [us].
+#define MAX_RETRY_DELAY 1000000  ///< Maximal allowed delay of procedure retry [us].
+
+static void timeout_timer_retry(void);
+
+static uint32_t           m_timeout = NRF_802154_PRECISE_ACK_TIMEOUT_DEFAULT_TIMEOUT;  ///< ACK timeout in us.
+static nrf_802154_timer_t m_timer;                                                     ///< Timer used to notify when we are waiting too long for ACK.
```
Comment (b-gent) [inline src/mac_features/nrf_802154_precise_ack_timeout.c]: Timer used to notify when the ACK frame is delayed for too long.

(not sure if my understanding is correct, if not , just leave as is)

File: src/nrf_802154_const.h [hunk 3]
```diff
@@ -94,16 +94,16 @@
 #define SHORT_ADDRESS_SIZE    2    ///< Size of Short Mac Address
 #define EXTENDED_ADDRESS_SIZE 8    ///< Size of Extended Mac Address
 #define FCS_SIZE              2    ///< Size of FCS field
-#define ACK_LENGTH            5    ///< Length of ACK frame
```
Comment (b-gent) [inline src/nrf_802154_const.h]: Please add full stops everywhere.

(unless merging this is time-critical, it's not a big deal but a stylistic issue)

File: src/nrf_802154_const.h [hunk 4]
```diff
@@ -94,16 +94,16 @@
...
+#define IMM_ACK_LENGTH        5    ///< Length of ACK frame
 #define MAX_PACKET_SIZE       127  ///< Maximal size of radio packet
 
 #define TURNAROUND_TIME       192UL                         ///< aTurnaroundTime [us]
 #define CCA_TIME              128UL                         ///< aCcaTime [us]
 #define UNIT_BACKOFF_PERIOD   (TURNAROUND_TIME + CCA_TIME)  ///< aUnitBackoffPeriod [us]
 
-#define PHY_US_PER_SYMBOL     16  ///< Duration of single symbol in microseconds [us]
+#define PHY_US_PER_SYMBOL      16 ///< Duration of single symbol in microseconds [us]
 #define PHY_SYMBOLS_PER_OCTET  2  ///< Number of symbols in single byte (octet)
```
Comment (b-gent) [inline src/nrf_802154_const.h]: a single byte

File: src/nrf_802154_const.h [hunk 5]
```diff
@@ -94,16 +94,16 @@
 #define SHORT_ADDRESS_SIZE    2    ///< Size of Short Mac Address
 #define EXTENDED_ADDRESS_SIZE 8    ///< Size of Extended Mac Address
 #define FCS_SIZE              2    ///< Size of FCS field
-#define ACK_LENGTH            5    ///< Length of ACK frame
+#define IMM_ACK_LENGTH        5    ///< Length of ACK frame
 #define MAX_PACKET_SIZE       127  ///< Maximal size of radio packet
```
Comment (b-gent) [inline src/nrf_802154_const.h]: Maximum size of a radio packet

File: src/nrf_802154_const.h [hunk 6]
```diff
@@ -94,16 +94,16 @@
...
 
 #define TURNAROUND_TIME       192UL                         ///< aTurnaroundTime [us]
 #define CCA_TIME              128UL                         ///< aCcaTime [us]
 #define UNIT_BACKOFF_PERIOD   (TURNAROUND_TIME + CCA_TIME)  ///< aUnitBackoffPeriod [us]
 
-#define PHY_US_PER_SYMBOL     16  ///< Duration of single symbol in microseconds [us]
+#define PHY_US_PER_SYMBOL      16 ///< Duration of single symbol in microseconds [us]
```
Comment (b-gent) [inline src/nrf_802154_const.h]: a single symbol

File: src/nrf_802154_core.c [hunk 7]
```diff
@@ -177,12 +177,12 @@ static rx_buffer_t * mp_current_rx_buffer;
 static rx_buffer_t * const mp_current_rx_buffer = &nrf_802154_rx_buffers[0];
 #endif
 
-static uint8_t         m_ack_psdu[ACK_LENGTH + 1]; ///< Ack frame buffer.
-static const uint8_t * mp_tx_data;                 ///< Pointer to data to transmit.
-static uint32_t        m_ed_time_left;             ///< Remaining time of current energy detection procedure [us].
-static uint8_t         m_ed_result;                ///< Result of current energy detection procedure.
+static uint8_t         m_ack_psdu[IMM_ACK_LENGTH + 1]; ///< Ack frame buffer.
+static const uint8_t * mp_tx_data;                     ///< Pointer to data to transmit.
```
Comment (b-gent) [inline src/nrf_802154_core.c]: to the data

File: src/nrf_802154_core.c [hunk 8]
```diff
@@ -177,12 +177,12 @@ static rx_buffer_t * mp_current_rx_buffer;
 static rx_buffer_t * const mp_current_rx_buffer = &nrf_802154_rx_buffers[0];
 #endif
 
-static uint8_t         m_ack_psdu[ACK_LENGTH + 1]; ///< Ack frame buffer.
-static const uint8_t * mp_tx_data;                 ///< Pointer to data to transmit.
-static uint32_t        m_ed_time_left;             ///< Remaining time of current energy detection procedure [us].
-static uint8_t         m_ed_result;                ///< Result of current energy detection procedure.
+static uint8_t         m_ack_psdu[IMM_ACK_LENGTH + 1]; ///< Ack frame buffer.
```
[the remark this answers is not in the document]
Comment (KaSroka) [reply src/nrf_802154_core.c]: They're using `Ack` in 2015 version of 802.15.4 specification.

**Pull request merged**